scipy/_lib/tests/test__util.py

```diff
@@ -302,7 +305,7 @@ def test_policy(self):
         with pytest.raises(ValueError, match=msg):
             _contains_nan(data, nan_policy="nan")
 
-    def test_contains_nan_1d(self):
+    def test_contains_nan(self):
```

Comment:
Just moved tests with non-numeric input into a separate function so they could be skipped.